packages/stream-model-instance-handler/src/model-instance-document-handler.ts

```diff
@@ -268,7 +280,7 @@ export class ModelInstanceDocumentHandler implements StreamHandler<ModelInstance
 
     // Now validate the relations
     await this._validateRelationsContent(ceramic, model, content)
-    if (!genesis && payload) {
+    if (!genesis && payload && !hasContent) {
```

Comment:
I'm trying to figure out how to make the behavior here more obvious to new readers of the code.  It's not immediately clear to me what `hasContent` means or why it affects the behavior of whether or not we should validate locked fields.  Can we at least update the doccomments for this function to include a description of the field and how it's used?

Maybe also consider naming the field `hasExistingContent` or something to distinguish that we're talking about whether the stream already had content before this update?

And maybe an inline comment here explaining that if this is a deterministic stream getting its content set for the first time we want to allow the immutable fields to be set to their initial value?

Reply:
I guess another possibility is renaming `hasContent` to simply `enforceImmutableFields` or something like that, then have a comment explaining the reasoning for how this gets set up in `_applySigned` at the call site?

Reply:
I did a mix of the suggestions by changing the name and also updating the function documentation